File: p028.py
def f(n):
    l = [1] 
    curr = 2 
    for delta in range(2, n, 2): 
        for i in range(curr + delta, curr + delta * 5, delta):
            l.append(i - 1)
        curr = l[-1] + 1
    return sum(l)

print f(1001)

File: go/src/p028/p028.go
package main

import (
    "fmt"
)

func main() {
    fmt.Println(sum(1001))
}

func sum(n int) (int) {
    total := 1
    for delta, curr := 2, 2; delta < n; delta += 2 {
        for i := curr + delta; i < curr + delta * 5; i += delta {
            total += i - 1
        }
        curr += delta * 4
    }
    return total
}
